pynetbox/core/endpoint.py

```diff
@@ -231,22 +236,24 @@ def filter(self, *args, **kwargs):
         if args:
             kwargs.update({"q": args[0]})
 
-        if not kwargs:
-            raise ValueError("filter must be passed kwargs. Perhaps use all() instead.")
         if any(i in RESERVED_KWARGS for i in kwargs):
             raise ValueError(
                 "A reserved kwarg was passed ({}). Please remove it "
                 "and try again.".format(RESERVED_KWARGS)
             )
-
+        limit = kwargs.pop("limit") if "limit" in kwargs else 0
+        offset = kwargs.pop("offset") if "offset" in kwargs else None
+        if limit == 0 and offset is not None:
+            raise ValueError("offset requires a positive limit value")
         req = Request(
             filters=kwargs,
             base=self.url,
             token=self.token,
             session_key=self.session_key,
             http_session=self.api.http_session,
             threading=self.api.threading,
-            limit=kwargs.get("limit", 0),
```

Comment:
I didn't check this, but if limit were in kwargs, wouldn't it also be passed as ```filters```? I think I fixed that with my approach - not sure it matters..